network/Message.cpp

```diff
@@ -594,6 +594,19 @@ Message ServerLobbyUpdateMessage(const MultiplayerLobbyData& lobby_data) {
     return Message(Message::LOBBY_UPDATE, os.str());
 }
 
+Message ChatHistoryMessage(const std::vector<std::reference_wrapper<const ChatHistoryEntity>>& chat_history) {
```

Comment:
What advantage does this new message have over sending a series of chat messages to the joining player with appropriate timestamps to send the the recent chat history...?

Reply:
https://github.com/freeorion/freeorion/pull/1815#discussion_r145879049
> Also I have no idea about implementation. History message cann't have an player id because player could disconnect before message will be shown so player name is mandatory. But chat should have an player id as it used for coloring.
I think it better to add new message type for chat history.

It is quicker than series of messages on TCP.